api/provider/provider.go

```diff
@@ -43,9 +44,9 @@ type Provider interface {
 
 	LogStream(app string, w io.Writer, opts structs.LogStreamOptions) error
 
-	ReleaseDelete(app, id string) (*structs.Release, error)
+	ReleaseBatchDelete(app, buildID string) error
```

Comment:
Would prefer to see it left as `ReleaseDelete` and the controller can call it multiple times. There should not be very many releases per build in most situations.